src/app/parcels/parcelsTable/ParcelsPage.tsx

```diff
@@ -117,6 +120,41 @@ const ParcelsPage: React.FC<{}> = () => {
         })();
     }, []);
 
+    function getValueBoundaries<DataType>(values: (DataType | null)[]): number[] {
+        const valueSet = new Set(values);
+
+        const boundaries: number[] = [];
+        valueSet.forEach((value) => {
+            if (value === values[0]) {
+                return;
+            }
+
+            const indexOfFirstOccurrence = values.indexOf(value);
+            if (indexOfFirstOccurrence === -1) {
+                //
+            } else {
+                boundaries.push(indexOfFirstOccurrence);
+            }
+        });
+        return boundaries;
+    }
+
+    const searchForPackingSlotBreakPoints = (parcelsTableRows: ParcelsTableRow[]): number[] => {
+        const packingSlotValues = parcelsTableRows.map((row) => {
+            return row.packingSlot;
```

Comment:
you don't need the `{return ...}` here. you can just do `.map((row) => row.packingSlot)` and similarly elsewhere.